pkg/cmd/dedicated/deregister/deregister.go

```diff
@@ -0,0 +1,304 @@
+	clusterList                   []clustersmgmtv1.Cluster
+	selectedCluster               clustersmgmtv1.Cluster
+	existingMachinePoolList       []clustersmgmtv1.MachinePool
+	selectedClusterMachinePool    clustersmgmtv1.MachinePool
+	requestedMachinePoolNodeCount int
+	accessKafkasViaPrivateNetwork bool
+	// newMachinePool                clustersmgmtv1.MachinePool
+
+	f *factory.Factory
+}
+
+// list of consts should come from KFM
+const (
+	clusterReadyState      = "ready"
+	fleetshardAddonId      = "kas-fleetshard-operator"
+	strimziAddonId         = "managed-kafka"
+	fleetshardAddonIdQE    = "kas-fleetshard-operator-qe"
+	strimziAddonIdQE       = "managed-kafka-qe"
+	billingModelEnterprise = "enterprise"
+)
+
+func NewDeRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "deregister-cluster",
+		Short:   f.Localizer.MustLocalize("dedicated.registerCluster.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.registerCluster.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.registerCluster.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runDeRegisterClusterCmd(opts)
+		},
+	}
+
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.clusterManagementApiUrl, "cluster-mgmt-api-url", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterMgmtApiUrl.description"))
+	flags.StringVar(&opts.accessToken, "access-token", "", f.Localizer.MustLocalize("dedicated.registercluster.flag.accessToken.description"))
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterId.description"))
+
+	return cmd
+}
+
+func runDeRegisterClusterCmd(opts *options) (err error) {
+	// Set the base URL for the cluster management API
+	err = setListClusters(opts)
+	if err != nil {
+		return err
+	}
+	if len(opts.clusterList) == 0 {
+		return opts.f.Localizer.MustLocalizeError("dedicated.registerCluster.run.noClusterFound")
+	}
+	// TO-DO if client has supplied a cluster id, validate it and set it as the selected cluster without listing getting all clusters
+	if opts.selectedClusterId == "" {
+		err = runClusterSelectionInteractivePrompt(opts)
+		if err != nil {
+			return err
+		}
+	} else {
+		for i := range opts.clusterList {
+			cluster := opts.clusterList[i]
+			if cluster.ID() == opts.selectedClusterId {
+				opts.selectedCluster = cluster
+			}
+		}
+	}
+
+	kafkas, err := getkafkasInCluster(opts)
+	if err != nil {
+		return err
+	}
+
+	err = deleteKafkasPrompt(opts, &kafkas)
+	if err != nil {
+		return err
+	}
+
+	err = deregisterClusterFromKasFleetManager(opts)
+	if err != nil {
+		return err
+	}
+
+	err = removeAddonsFromCluster(opts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getClusterList(opts *options) (*clustersmgmtv1.ClusterList, error) {
```

Comment:
there is a possibility of re-use here with what was done for registering command